server/map_rpc_server.go

```diff
@@ -171,6 +169,14 @@ func (t *TrillianMapServer) getLeavesByRevision(ctx context.Context, mapID int64
 	}, nil
 }
 
+func checkIndexSize(index []byte, hasher hashers.MapHasher) error {
+	if len(index) != hasher.Size() {
+		// TODO(drysdale): why does the leaf index need to be the same size as the tree hash?
```

Comment:
Drive-by: It's not an index, the API is misnamed. I just merged a PR that adds more doc to the proto.

Reply:
It is also an index, it's a large large number, but the hashed key value which gives the path to the leaf by definition is also the ordinal number of that leaf.

Strictly, the leaf index doesn't need to be the same size at the tree root hash (e.g. if the index was calculated using SHA512, but the tree hashes were SHA256), that's just how it is currently.

Reply:
Yes - true but people who come to this from the log code might expect an 'index' API to behave similarly, which isn't really the case.

Reply:
I agree with @Martin2112 that `index` is an unhelpful name, but it's at least a name that's consistent with all the current code...

Reply:
I think it's too late to change it now.

Reply:
The key requirement is that all leaves in the map have the same `index` length.